connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -539,6 +539,15 @@ class SparkConnectPlanner(session: SparkSession) {
    * @return
    */
   private def transformScalarFunction(fun: proto.Expression.UnresolvedFunction): Expression = {
+    if (fun.getFunctionName == "product") {
```

Comment:
My concern on such way is it looks too specialized. Not sure how many such case we will have, but at least can we pull this `if and if` to a validation function?

Reply:
good point, I think we will have more similar cases, let me add a new function for it.